src/main/java/com/tabnine/intellij/completions/TabNineCompletionContributor.java

```diff
@@ -53,6 +56,12 @@ public void fillCompletionVariants(
     if (originalMatcher.getPrefix().length() == 0 && completions.results.length == 0) {
       return;
     }
+
+    if (suggestionsModeService.getSuggestionMode() == SuggestionsMode.HYBRID
+        && Arrays.stream(completions.results).anyMatch(Completion::isSnippet)) {
+      return;
+    }
```

Comment:
usage of the hybrid mode - dont render popup completions if they contain snippets

Reply:
why not remove the snippet completions as you did in the InlineCompletionHandler?

Reply:
Good catch 🙂
I saw it on dima's branch in vscode and thought so as well, but there's a reason for doing so - 

When you get a snippet and type the same prefix as the snippet, you'll get both inline and popup completions, and we don't want that (at least currently)

Reply:
got it! thanks :)